cluster-up/cluster/kind-k8s-sriov-1.17.0/config_sriov.sh

```diff
@@ -12,6 +12,135 @@ WORKER_NODE_ROOT="${CLUSTER_NAME}-worker"
+# until the command returns an empty string or until timeout
+function retry {
+  local -r tries=$1
+  local -r wait_time=$2
+  local -r action=$3
+  local -r wait_message=$4
+
+  local result=$(eval $action)
+  for i in $(seq $tries); do
+    if [[ -z $result ]] ; then
+      echo "[$i/$tries] $wait_message"
+      sleep $wait_time
+      result=$(eval $action)
+    else
+      return 0
+    fi
+  done
+
+  return 1
+}
+
+function wait_for_daemonSet {
+  local name=$1
+  local namespace=$2
+  local required_replicas=$3
+
+  if [[ $namespace != "" ]];then
+    namespace="-n $namespace"
+  fi
+
+  if (( required_replicas < 0 )); then
+      echo "DaemonSet $name ready replicas number is not valid: $required_replicas" && return 1
+  fi
+
+  local -r tries=30
+  local -r wait_time=10
+  wait_message="Waiting for DaemonSet $name to have $required_replicas ready replicas"
+  error_message="DaemonSet $name did not have $required_replicas ready replicas"
+  action="_kubectl get daemonset $namespace $name -o jsonpath='{.status.numberReady}' | grep -w $required_replicas"
+
+  retry "$tries" "$wait_time" "$action" "$wait_message" && return  0
+  echo $error_message && return 1
+}
+
+function wait_pod {
+  local namespace=$1
+  local label=$2
+
+  local -r tries=60
+  local -r wait_time=5
+
+  local -r wait_message="Waiting for pods with $label to create"
+  local -r error_message="Pods with  label $label at $namespace namespace found"
+
+  if [[ $namespace != "" ]];then
+    namespace="-n $namespace"
+  fi
+
+  if [[ $label != "" ]];then
+    label="-l $label"
+  fi
+
+  local -r action="_kubectl get pod $namespace $label -o custom-columns=NAME:.metadata.name --no-headers"
+
+  retry "$tries" "$wait_time" "$action" "$wait_message" && return  0
+  echo $error_message && return 1
+}
+
+function wait_k8s_object {
+  local -r object_type=$1
+  local -r name=$2
+  local namespace=$3
+
+  local -r tries=60
+  local -r wait_time=3
+
+  local -r wait_message="Waiting for $object_type $name"
+  local -r error_message="$object_type $name at $namespace namespace found"
+
+  if [[ $namespace != "" ]];then
+    namespace="-n $namespace"
+  fi
+
+  local -r action="_kubectl get $object_type $name $namespace -o custom-columns=NAME:.metadata.name --no-headers"
+
+  retry "$tries" "$wait_time" "$action" "$wait_message" && return 0
+  echo $error_message && return  1
+}
+
+
+function is_taint_absence {
+  local -r taint=$1
+
+  result=$(_kubectl get nodes -o jsonpath="{.items[*].spec.taints[?(@.effect == \"$taint\")].effect}")
+  if [[ -z $result ]]; then 
+    echo "not-present"
+  fi
+}
+
+function wait_for_taint_absence {
+  local -r taint=$1
+
+  local -r tries=60
+  local -r wait_time=5
+
+  local -r wait_message="Waiting for $taint taint absence"
+  local -r error_message="Taint $taint $name did not removed"
+  # '-v'  flag propagates the string that echo'ed by 'is_taint_absence'
+  local -r action="is_taint_absence $taint | grep -i $taint -v"
```

Comment:
$ function bla() {
> x=$(echo 123)
> }

$ bla
$ bla | grep 123 -v

how come -v propagate the var that is encapsulated in result var ?
what do i miss ?

btw the right use is the cause the return code in the function, but i wont stall on it

Reply:
I see what is the problem now thanks.
I pushed new commit with different approach to solve it.